### DISCUSSION TITLE
Compiling Rebol3 on Haiku OS fails at linker stage

### DISCUSSION THREAD

**Discussion by fvanzeveren** [Oldes/Rebol3#66] [Q&A] [answered]
Hello
I would like to compile Rebol3 from the sources on Haiku OS Beta 4.
All goes well until linker stage where the following error occurs:

```
/packages/gcc-2.95.3_2017_07_20-2/.self/develop/tools/i586-pc-haiku/bin/ld: make/tmp/clang-x86/rebol3-bulk-haiku-x86/boot/home/git/Rebol3-master/src/core/a-lib.c.o: unrecognized relocation (0x2b) in section `.text'
/packages/gcc-2.95.3_2017_07_20-2/.self/develop/tools/i586-pc-haiku/bin/ld: final link failed: Bad value
collect2: ld returned 1 exit status
clang-12: error: linker (via gcc) command failed with exit code 1 (use -v to see invocation)
```

What should I do to fix this issue?

Thank you.

François

**Comment by Oldes** [comment, marked as answer]
I don't know... I used cross compilation using a Docker image
https://github.com/Oldes/Rebol-Docker/blob/master/.github/workflows/build-haiku-rebol.yml
It works ok: https://github.com/Oldes/Rebol-Docker/actions/runs/4730560363

**Reply by fvanzeveren** [reply]
Thanks for pointing me to the docker image. I can now make a Haiku build from my Debian workstation! Great!

**Comment by rebolek** [comment]
I was able to compile Rebol3 on Haiku x86, I just needed proper version of clang

**Reply by fvanzeveren** [reply]
Thanks you for feedback. Can you tell me which version to use, because in the haïku repo,  i see several ones. I tried with the last one available (12) without success. 🧐

**Reply by Oldes** [reply]
To be honest, I don't know... Haiku is not my primary OS. For me is important, that I am able to make a build for it using the Docker image, because I can automatize such builds.

**Comment by Oldes** [comment]
According [this post](https://stackoverflow.com/a/47725251/494472), the reason is, that the linker (`ld`) is older then the compiler used.

**Comment by fvanzeveren** [comment]
Hello

After some trial and errors, using llvm6_x86 does the job!

I am now able to compile Rebol3 Core and Bulk on my Haiku workstation !!

Note: building Rebol3/base fails with the following error:
[SISKIN] ** Error: Failed to process::include-rebol-base
 [SISKIN] ** Error: Command not handled: 1

Regards

**Reply by Oldes** [reply]
Fine.. I just pushed fix for the Rebol/Base: https://github.com/Oldes/Rebol3/commit/70e244a61340f5d7b442a3aec099ea2762bd85f2
